x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/api.ts

```diff
@@ -126,7 +127,23 @@ export const fetchRules = async ({
  * @throws An error if response is not OK
  */
 export const fetchRuleById = async ({ id, signal }: FetchRuleProps): Promise<Rule> =>
-  KibanaServices.get().http.fetch<Rule>(DETECTION_ENGINE_RULES_URL, {
+  pureFetchRuleById({ id, http: KibanaServices.get().http, signal });
+
+/**
+ * Fetch a Rule by providing a Rule ID
```

Comment:
Why is this function needed? Is it so we can use `withOptionalSignal`? Just asking because I'm curious, though it might be good to add a line in the comment that touches on the difference between `fetchRuleById` and `pureFetchRuleById`.

Reply:
Just saw this one: we need this form of the function in order to use it with the `useAsync` hook.